src/main/java/ch/securify/decompiler/evm/OpCodes.java

```diff
@@ -209,6 +209,30 @@ public static boolean isInvalid(int opcode) {
 		return "INVALID".equals(getOpName(opcode)) || opcode == INVALID;
 	}
 
+	/**
+	 * Indicate whether the opcode terminates a basic block
+	 * Ignores the case of JUMPI for which it depends on the CFG
```

Comment:
Why does JUMPI not end a block?

Reply:
In the code there is a check on `controlFlowGraph.get(rawInstruction.offset).contains(ControlFlowDetector.DEST_ERROR)` when a `JUMPI` is encountered, meaning that Securify currently does not assume that this is the end of the block just because of this opcode. Ignoring `JUMPI` in this function was a way to refactor the code without changing the semantics of the basic blocks Securify currently builds.